status/status.go

```diff
@@ -257,20 +257,25 @@ func (s *Status) ReportSnapshotStatus(ctx context.Context, reporter ReporterInte
 		}
 		testReport, err := s.generateTestReport(ctx, *integrationTestStatusDetail, snapshot)
 		if err != nil {
-			_ = WriteSnapshotReportStatus(ctx, s.client, snapshot, srs) // try to write what was already written
+			if err := WriteSnapshotReportStatus(ctx, s.client, snapshot, srs); err != nil { // try to write what was already written
+				return fmt.Errorf("failed to write snapshot report status metadata: %w", err)
```

Comment:
Thinking about it, we are losing information about what happened in generateTestReport; we should return both errors, via error.Join